### PR TITLE
fix(polymorphic): update of type of a polymorphic key works

### PR DESCRIPTION
**Pull request by VincentMolinie**
Updating a fragment to something that should be another fragment type doesn't work. I just make sure that when setting a fragment value the type is the same otherwise recreate the fragment

### CODE REVIEW

**Comment by VincentMolinie** [comment]
@knownasilya needs your approval here if you got time 👼

File: addon/record-data.js [hunk 1]
```diff
@@ -149,7 +150,7 @@ export default class FragmentRecordData extends RecordData {
     } else if (isFragment(value)) {
       // A fragment instance was given, so just replace the existing value
       fragment = setFragmentOwner(value, record, key);
-    } else if (!fragment) {
+    } else if (!fragment || (getActualFragmentType(declaredModelName, options, value, record) !== fragment.constructor.modelName)) {
```
Comment (knownasilya) [inline addon/record-data.js:153]: I think we need a comment here, or maybe put this into a descriptive variable so we know what this is doing in the future.

**Review by knownasilya — CHANGES_REQUESTED** [review]
Small change, otherwise it's 🚀

**Comment by VincentMolinie** [comment]
Done @knownasilya thanks again for your quick reply 🙏

**Review by knownasilya — CHANGES_REQUESTED** [review]
Tiny change

**Review by knownasilya — APPROVED** [review]

**Comment by VincentMolinie** [comment]
Thank you so much @knownasilya 🙏

**Pull request merged**